intel_npu_acceleration_library/compiler.py

```diff
@@ -174,9 +204,85 @@ def optimize_llama_attention(
     return None
 
 
+@module_optimization
+def optimize_phi3_MLP(
+    name: str, layer: torch.nn.Module
+) -> Union[torch.nn.Module, None]:
+    """Optimize Phi-3 MLP block.
+
+    Args:
+        name (str): Module name
+        layer (torch.nn.Module): Original Module
+
+    Returns:
+        Union[torch.nn.Module, None]: optimized Phi-3 module
+    """
+    if layer.__class__.__name__ == "Phi3MLP":
+        return layer
+    return None
+
+
+@module_optimization
+def weights_quantization(
+    name: str, layer: torch.nn.Module
+) -> Union[torch.nn.Module, None]:
+    """Apply weights quantization.
+
+    Args:
+        name (str): Layer name
+        layer (torch.nn.Module): Original torch.nn.Linear module
+
+    Raises:
+        RuntimeError: unsupported quantization bits
+
+    Returns:
+        None: Returns None
+    """
+    if isinstance(layer, WeightOnlyLinear):
+        if (layer.bits == 4) or (layer.bits == 8):
+            layer.forward = partial(forward, layer)
+        else:
+            raise RuntimeError(f"Unsupported quantization bits: {layer.bits}")
+    return None
+
+
+def forward(self, input):
+    """Override forward method for WeightOnlyLinear class.
+
+    Args:
+        input: The input tensor.
+
+    Returns:
+        torch.Tensor: The output tensor.
+    """
+    if self.bits == 4:
+        # Unpack the int4 values
+        lower_int4 = self.qweight & 0x0F
+        lower_int4 = lower_int4 - (lower_int4 & 0x8) * 2
+        upper_int4 = (self.qweight >> 4) & 0x0F
+        upper_int4 = upper_int4 - (upper_int4 & 0x8) * 2
+
+        w = torch.stack((lower_int4, upper_int4), dim=2)
+        w = w.contiguous().view(self.qweight.shape[0], -1)
+
+    elif self.bits == 8:
+        w = self.qweight.view(torch.int8)
+
+    output = (
+        torch.nn.functional.linear(input.to(torch.float16), w.to(torch.float16), None)
+        * self.scales.T
+    )
+
+    if self.bias:
+        return output + self.bias
+
+    return output
```

Comment:
This is only needed with the `.to` method. Can you make it run only with that?

Reply:
with the ```.to()``` method, the type ```NPUModuleWrapper``` is created, therefore this will never have an effect on the block as it is not a linear layer we are working with after the initial ```quantize_model()``` which still just returns the model of type ```NPUModuleWrapper```

Reply:
This functionality has been added